### ISSUE TITLE
Colaboration?

### ISSUE THREAD

**Issue by dmkpoznan** [AngelDTF/OpenParsec#2]
Hi,

I made a lot of changes to your code implementing mouse (GCMouse), keyboard (GCKeyboard), touch & pencil events, auto-login, username/password saving and more. 

For now code is on my local repository (not published yet), but I plan to publish it on github.

The question is about colaboration (are you interested?) and if you plan to develop anything more in this project?

Best regards,
Dawid

PS For now it's fully working remote desktop and I use it as my laptop on M1 iPad connected to external monitor - it's working flawless! Thank you for inspiring me to do that.

**Comment by AngelDTF** [comment]
Hello, I’m glad I was able to give you the inspiration to do further work on the project!

Regarding your questions, yes I still plan to develop more for this project in the future, I’ve just put a temporary hold on making further progress due to a backup of other projects and to focus on my health, as me not taking care of myself properly has caused me to go to the ER not too long ago, and I don’t want that to happen again.

About collaboration, I am interested, however I’d like to see some of the changes you made personally before making any big decisions.

Would you be open to moving to a different chatting platform for easier discussion? If you are and have Discord, send me your name and tag number, and I could add you for further conversations. If you don’t have Discord, we could continue here or on another platform we both have.

**Comment by kibabyte** [comment]
> For now code is on my local repository (not published yet), but I plan to publish it on github.

That code would be a godsend!

**Comment by dmkpoznan** [comment]
@AngelDTF please email me here: 07kopac_baczny@icloud.com (it's temporary address hiding my email).

Best regards

PS Sneak peak of current version (please note, that I recorder it using LTE only to proof that this is not only LAN-working version, so this is main source of latency around 30+ms):

https://www.icloud.com/photos/#09fcSeXlH1bl-emwJrkPOzHVQ

What's working?

* session persistance
* autoconnect
* sleep/resume connection on hiding window
* keyboard with repeat (this was not obvious to achieve)
* mouse with gestures (like two fingers scrolling)
* touch with gestures (like two fingers scrolling)
* pencil touches
* full screen (without overlays etc.)
* multimonitor (thanks to stage manager, M1 and iOS 16)

As you can see it can replace 4K Mac. I connect to basic M1 mac mini (for test purposes) and it's working flawless.

**Comment by kibabyte** [comment]
@dmkpoznan Amazing what you already were able to achieve! If you want someone to beta test it for you, hit me up.

**Comment by Jiankun-Huang** [comment]
Same, I'll be happy to test it!

**Comment by borjandev** [comment]
Would be happy to contribute, any way to test it? Thanks

**Comment by enchained** [comment]
I'd like to test it too if possible. Can it work with older hardware like 6th gen iPad (2018)?

**Comment by JAIL10276** [comment]
I'm so desperate, I've been checking this every single second that I can. I hope you find yourself in good health tho @AngelDTF

**Comment by JAIL10276** [comment]
I would also be down to beta test it.

**Comment by TonttuHomo** [comment]
Potential tester here too!! 😊

**Comment by Mannilie** [comment]
Happy to help too! I finally got OpenParsec working in my test environment. Looking great so far, just unable to use touch inputs.

**Comment by LeaveNhA** [comment]
Oops, I've made some contributions, before reading here too.

* TFA (Two-Factor Authentication support)
* Session save, so you don't have to log-in again and again.
* Some fixes over some edge-cases like if API will not return any host information (in case of no hosts are available to you).

**Comment by michaelsit91** [comment]
I don’t mind helping to test the app too

**Comment by Rafa2212** [comment]
Hello, any updates?

**Comment by AngelDTF** [comment]
Have been waiting on an email for weeks now, haven't gotten anything back yet. I would've responded sooner but my medical condition has been making it difficult to go on my PC sometimes. @LeaveNhA if you'd like to make a pull request, I'd be happy to look through to see what can be added to the project.

**Comment by LeaveNhA** [comment]
@AngelDTF, ofcourse. I would love to create a PR. Not much, but honest work. 😂

**Comment by dmkpoznan** [comment]
Hi all,

I got a lot of emails and questions (even here) asking about testing my version. I made big step forward with functionality (implementing even battery saving - reducing framerate to ultra-low values on inactivity). I redesigned UI, animations, added screenshots for thumbnails of remote screens and still have backlog for future features. I also added multi-window connections (thanks to iOS 16 and Pro iPads), FPS switching and much more.

I even renamed project (from OpenParsec to PocketRocket)...

You can ask why I'm still not sharing it..? The reason is based in license. I contacted (as our company) with Parsec (formally owned by Unity!) and I'm trying to figure out licensing model for iOS client. The name including their trademark ("Parsec") could be potienally problematic (this is the reson why I decided to rename it). But the main problem could be sharing code with SDK library without their permision. So I decided to hold everything untill I can speak with someone from Unity. For now I ended up contacting somebody from sales who are not understanding the case (they offered us selling SDK for 1 milion dolars in licensing model, then she thought that we want to develop for parsec offering our development team... what a communication mess) and I'm trying to reach to some management staff.

Untill things with license will not clarify I'm not going to break license of parsec, so basicly I can't share solution in any way (in fact I would love to share it via TestFlight right now).

The another problem is that we all are using some old version of SDK bundle that was available some time ago on github of parsec. Probably after selling Parsec to Unity they internally decided to remove it.

If anyone have some contact(s) to Unity team or have idea how to break this barier let me know. I'm out of ideas and still waiting for reply from Unity (in fact they are replying after 1-2 weeks after my email).

Best regards

PS Some screenshots this time:

![IMG_0111](https://user-images.githubusercontent.com/2170338/225244034-15db63f9-6fb9-4dfa-af13-7468893f52e1.jpg)
![IMG_0112](https://user-images.githubusercontent.com/2170338/225244041-20b16cf7-8083-4a15-93af-d74c63740e28.jpeg)
![IMG_0113](https://user-images.githubusercontent.com/2170338/225244042-d943f696-cdd1-481e-92f6-3a49cc0f9d72.jpg)
![IMG_0114](https://user-images.githubusercontent.com/2170338/225244046-98511109-49ce-4f81-8668-2aa908b17c15.jpg)

**Comment by LeaveNhA** [comment]
I do not wish to participate in a project that has been renamed and is now commercial and closed-source.

In the spirit of open-source, I propose that we engage in discussions to gather various perspectives and needs in order to find solutions and create a roadmap that allows for free and open contributions. Though the current state of the project satisfies my personal needs of having a side monitor accessible regardless of my location or activity, others may have different intentions such as using it for gaming. I am open to suggestions and ideas, and encourage anyone to contact me without hesitation as this is GitHub, not Microsoft.

**Comment by dmkpoznan** [comment]
It’a a little bit funny, that you are talking about “true open-source spirit” from perspective of using (possibly illegal) SDK of commercial product from other company. The code that all of us written here is more or less only some box for existing solution. And here I should stop but let me explain deeply all of reasons, of my decisions:

1. Naming “OpenParsec” solution, when Parsec is trademark break the law and this trademark. I strongly believe that this was not intentionally, but as owner of some software house I see this “detail” as real-world problem.
2. We want to be officially allowed to use Parsec (owned by Unity) SDK. For now we have only some unofficial SDK lib, that someone downloaded from their GitHub many months ago. They decided to not share SDK this way anymore. This is their product and we need to respect their decision. For now we are trying to reach someone, who can share with use SDK and licensing model. This is the case.
3. Telling “open-source” you mean this few line of code, that you or me have written? The biggest value is in SDK. Everything else doesn’t matter. And it’s commercial product. So we will never be “open-source”. If you like to contribute some code in “open-source spirit” go and write some Linux patches. 
4. I’m not telling that I’m not going to share code (in fact this was my first thought what should I do). But this could make difficult to convince Unity to share, somehow, their SDK. And also could prove, that we have unofficial access to their SDK library. And this is exactly what we don’t want. 

To sum up as a software developer you need to stop thinking about open-source and commercial as 0/1 things. World is not black and white. And licensing is real pain, but you can’t ignore it. For now we are dependent on Parsec, not the other way. So we need to ask and be patient waiting for answer.

**Comment by LeaveNhA** [comment]
Triggered, already? It's okay bro, calm down. You are the king, okay? You know everyting, including these things. (LOL)

Now, excuse us, if you don't have anything to bring in to the table.

**Comment by dmkpoznan** [comment]
Sorry, but You probably don't understand how real world works. And for now leave please leave this topic to not create more miss-information. 

To sum up:

1. I would love to share for tests current build, but I can't for now because I could break Apple and Unity licensing (there is no other legal options than using TestFlight for now).

2. Providing code of my client could bring on table arguments from Unity side to not share SDK with us. We (I) don't want this!

3. I will probably share code after beeing allowed by Parsec/Unity to use their SDK (official way with licensing model, not in pirate model like "download SDK from some shitty-zip-site, add to project, build and deploy on your device". This way is maybe good for some jailbrake apps, but we don't need anything like that (for example JIT).

In fact I can only be triggered by someone stupidity, so don't continue this.

**Comment by LeaveNhA** [comment]
Everything I bring, or I said can be verified/falsified in any time, so, please stop crying. I understand enough for me. So, mind your business.

I'm advocating for the project to remain open-source and available for collaboration in the spirit of the open-source community. I'm more focused on the development and potential uses of the project, rather than the licensing and legal implications. I don't think they will sue me or any of us for using it a couple of hours (in total, yeah), PERSONALLY.

I didn't like your tone, your toxicity. If you hadn't done this, we could simply discuss, you could give us ideas (if you have some, I really am not sure you have, now) but all you do is sharing some content of (ILLEGAL?) application's development and it's features. I or any of us don't need any of these. I need simply a software to solve my real-world problem. Now, if we can develop it, we can use it in our daily lives.


PS: I'm so sorry for you.

**Comment by kibabyte** [comment]
To maybe help with clarification: [This](https://github.com/FlavioFS/ParsecSoda#license) is another app that integrates the ParsecSDK and is handling the Licensing similar to what I suggested.

**Comment by dmkpoznan** [comment]
I know ParsecSoda. But they are using windows SDK that is two years old! Even library that 	I found somewhere is older, than current official SDK documentation - so there is few differences. I still have some problems with SDK (like leaks of memory INSIDE Parsec library) that I can't fix without updating it. So I strongly believe, that the best way to have it on iOS is to get official way SDK.

Licensing of client is nothing important here, but parsec is something bigger. 

Probably I will share my solution with headers only, but this will be useless for most of you until you find exactly same version of SDK that I'm using. And I'm not going to share it to have a chance getting licensing for iOS client.

Parsec as protocol/library is amazing. I'm now using my computer that I left 1500km away and latency is as low as I can imagine.

**Comment by nomi-san** [comment]
Sure, their BUD protocol is amazing,

The old SDK could be found in any forked before Jul 2022. It's outdated since 29 Apr 2021, but still works fine. The official docs has been updated for new SDK since the end of 2022, but I have found this [old docs](https://parsec-v2.jake.museum/docs/sdk/).

There are a few things I hate about the old SDK
- FFMPEG support buggy
- 4:4:4 bug, could not be enabled
- New hardware buggy
- Symmetric NAT sometimes failed to establish P2P connection (on IPV6)
- Also memory leaks

Do you have the official SDK?
Hope for the private share 🙃

**Comment by dmkpoznan** [comment]
The old SDK is working great, but still have some bugs as you mentioned (I found even more problems, but this is long story).

I checked vdi-stream-client and I must say that they have only headers in their repo and if you want to run it you must have SDK library. So same story, different project.

Sorry to say, but still no response from Unity/Parsec about accessing official SDK.

For now I'm on my way to buy Apple Developer Account subscription & share "PocketRocket" on TestFlight (inviting you all to test it).

**Comment by Kodikuu** [comment]
> For now I ended up contacting somebody from sales who are not understanding the case (they offered us selling SDK for 1 milion dolars in licensing model, then she thought that we want to develop for parsec offering our development team... what a communication mess) and I'm trying to reach to some management staff.

This is the only way we license out the modern SDK now, as the form the SDK takes has changed.
The modern Parsec SDK is not the same as what you have seen and used, and will never be provided in a public format like the old one.

The old SDK was removed, and is no longer provided by us. Old copies may be used for existing projects, but there are no guarantees about how long it will continue to work. Per our licensing, we have no obligation to provide support or assistance with the old SDK, except where we have agreed prior, in writing.

There will be no update, bugfixes, or assistance of any kind with the old SDK.

We don't mind you using the name OpenParsec (see: Parsec Soda).

**Comment by kibabyte** [comment]
The people over at Parsec are not really interested in an iOS client. Which is weird when you consider, that they target businesses/teams. ![image](https://user-images.githubusercontent.com/97415279/229535484-3682b9e2-ecfc-4d0e-811b-08221111ac0a.jpeg)

**Comment by Kodikuu** [comment]
> The people over at Parsec are not really interested in an iOS client. Which is weird when you consider, that they target businesses/teams.

Oh hey, it's me.

I'm not the one that makes decisions on iOS, or would implement iOS. We _do_ want iOS, but we have _many_ other priorities, and adding an OS to our suite takes more effort than it may seem to you. We _won't_ be using Swift, for one. macOS _hosting_ alone has taken a very long time to iron out the quirks, and we're unlikely to entertain any new platforms at all until we're more or less done there. Another possibility is Linux hosting, and if we start on that/iOS, then the other will likely wait until after.

**Comment by LeaveNhA** [comment]
With all due respect, as far as I understand, you guys simply don't (and maybe won't) care about iOS. It's understandable. Apple, right? (Maybe other things between you fellow corp, dunno)

I just wanted to contribute, not much but honest work. And, after all the disrespectful words and intentions, I decided to make it a hobby to learn SDK and Parsec's awesome implementation more.

I've just added drag-and-slide functionality. I'll create a PR. If didn't, might found a mistake but you all can still access the source code in my repository.
I don't wanna piss of anyone fellas, I just want you guys to use this heavenly beautiful piece of software.

Hope you guys forgive me. 🙏🏻

Attention: @AngelDTF

**Comment by LeaveNhA** [comment]
I wish you all a good night, if it's possible: https://developer.apple.com/documentation/uikit/

**Comment by LeaveNhA** [comment]
Also, Keyboard support is available from now on! 🥳🎉

PS: Sorry again fellas, my mentality is a bit diverged from others. I cannot just sit and wait.

**Comment by AngelDTF** [comment]
> To sum up as a software developer you need to stop thinking about open-source and commercial as 0/1 things. World is not black and white. And licensing is real pain, but you can’t ignore it. For now we are dependent on Parsec, not the other way. So we need to ask and be patient waiting for answer.

Let me just start with the reason I started this project in the first place. It was because I got tired of waiting. I've been waiting for 5 years now (been using Parsec since 2018) and asked if they had an iOS client in the making. Multiple times I've been given both yes and no answers, to the point where I no longer believed they would do it themselves, at least not any time soon. And now that they're a part of Unity, I don't think it would be as simple as to request a license to use their API. Even if you could get it, you probably wouldn't be allowed to open source then, which leads into my next point. I open-sourced this project in an early state because in full honesty, I didn't think I would finish this project, but I was hoping it would give someone else the inspiration to either make additions or complete it themselves, and either have them send in a PR or just for it to live on through some fork of this repository.

> I do not wish to participate in a project that has been renamed and is now commercial and closed-source.
> 
> In the spirit of open-source, I propose that we engage in discussions to gather various perspectives and needs in order to find solutions and create a roadmap that allows for free and open contributions.

I agree with this whole-heartedly. I've seen people make their own iOS clients for Parsec but had no open-source code or really any method of taking a look at what's behind the scenes to make changes or necessary updates as people see fit. Some of these clients actually required a download through some sketchy third-party app store, which I'm not going to do since I'd appreciate not risking my credentials being stolen. All of this is what the inspiration was for me starting this project was in the first place, so that I could get possible constructive criticism, and give people something that they've also been waiting years for.

> It’a a little bit funny, that you are talking about “true open-source spirit” from perspective of using (possibly illegal) SDK of commercial product from other company.

There's nothing illegal about using previously open source files as long as the conditions of the license are still being met. The SDK isn't even being bundled in this repository, so the legal ramifications fall unto what the user decides to do with it. Since I'm personally using an unmodified copy of the old SDK and have not distributed it with this project, I have done nothing wrong.

> The old SDK was removed, and is no longer provided by us. Old copies may be used for existing projects, but there are no guarantees about how long it will continue to work. Per our licensing, we have no obligation to provide support or assistance with the old SDK, except where we have agreed prior, in writing.
> 
> There will be no update, bugfixes, or assistance of any kind with the old SDK.
> 
> We don't mind you using the name OpenParsec (see: Parsec Soda).

Fair enough. I assume Unity deals with all the licensing now since the acquisition. Hopefully the old API lasts long enough for this project to be useful for the next upcoming couple of years. That may just be a pipe dream though. Thanks for allowing the use of the name OpenParsec.

**Comment by LeaveNhA** [comment]
I just wanna say that It's an honor to work with you, and many of you who thinks freely!

**Comment by Aarav5255** [comment]
Can someone give me the ipa

**Comment by MikaPOINT** [comment]
> 

could i get invited to the testflight :)?

**Comment by kikucxj** [comment]
> how do i install it

**Comment by fm1069** [comment]
> Hey y'all,
> 
> Here's the most recent build of OpenParsec **(you'll need to sign it yourselves)**
> 
> https://www.dropbox.com/s/cxzzdq88tfy4y8m/OpenParsec.ipa?dl=0

hi，that`s link can not work anymore ,would you pls upload again,tks

**Comment by paulitaopa** [comment]
Why doesn't it work? 
![4E72C6B6-A42D-4E2B-AD7B-25DDAA929512](https://github.com/AngelDTF/OpenParsec/assets/122658020/861bb0a9-169d-4cae-a632-440bfc6094e0)

**Comment by hellosquishies** [comment]
I’m sorry but is there any controller support at this moment?
I got OpenParsec to work on mobile data for me on my iPhone 13 mini and it works pretty great but so far not a single game works with my DualShock 4 connected over BT.

**Comment by LeaveNhA** [comment]
No Controller support now, but it's in our road-map. Due to my brand-new company businesses and other open-source contributions originated my company stack are taking up all my time.

I believe, it will be implemented by the next month's end.

**Comment by LeaveNhA** [comment]
@AngelDTF, we need to discuss about a various things.

**Comment by hellosquishies** [comment]
Awesome work guys, keep it up!
Been waiting years for this iOS port and y'all are doing gods work

**Comment by LeaveNhA** [comment]
You do not need to wait anymore. Also, I'll create a guide for using it in daily workflow.

I can give some clue here that some of you fellow users are craving for, I believe. Like;
While using it for secondary monitor, which is my favorite usage for a quite while, I use [MouseHook](https://github.com/brgj/MouseHook) to make my mouse visible in the secondary monitor that I'm feeding to my phone or tablet (or my fiancé's gaming laptop). 

Also, please leave a thumbs-up for that guide because I was keep postponing it because of community might not need it.

**Comment by AngelDTF** [comment]
> @AngelDTF, we need to discuss about a various things. We can set-up a quick meeting, like, 15 min.

@LeaveNhA

Of course. When do you want to do this (I'm in [EDT time](https://time.is/EDT), free anywhere between 1PM to 11PM, but I can try to make earlier work if need be), and what method of communication would we be using?

**Comment by BLKKKBVSIK** [comment]
Hi.
Installed the latest build, and I'm very please with what I saw.
@LeaveNhA was speaking about drag/drop, but was are the basic gesture on iOS ? I couldn't make a simple tap to do a left click, also would be nice to have a gesture to zoom on the host screen.

**Comment by LeaveNhA** [comment]
> > @AngelDTF, we need to discuss about a various things. We can set-up a quick meeting, like, 15 min.
> 
> @LeaveNhA
> 
> Of course. When do you want to do this (I'm in [EDT time](https://time.is/EDT), free anywhere between 1PM to 11PM, but I can try to make earlier work if need be), and what method of communication would we be using?

Sorry for my late answer @AngelDTF,

I believe we can use Google Meet and plan it for later because of my heavy workload.

--

> Hi.
> Installed the latest build, and I'm very please with what I saw.
> @LeaveNhA was speaking about drag/drop, but was are the basic gesture on iOS ? I couldn't make a simple tap to do a left click, also would be nice to have a gesture to zoom on the host screen.

I believe it's working but it's not proven by any test/process. I personally use it for my daily work/life. Have you checked your permissions on Parsec end?

**Comment by BLKKKBVSIK** [comment]
> > Hi.
> > Installed the latest build, and I'm very please with what I saw.
> > @LeaveNhA was speaking about drag/drop, but was are the basic gesture on iOS ? I couldn't make a simple tap to do a left click, also would be nice to have a gesture to zoom on the host screen.
> 
> I believe it's working but it's not proven by any test/process. I personally use it for my daily work/life. Have you checked your permissions on Parsec end?

After more testing, it seems that I can tap for left click. My bad.
But using your phone to control a Windows PC, made the UI way to small, we need something to zoom on the host screen to tap at the right place.

**Comment by AngelDTF** [comment]
> Sorry for my late answer @AngelDTF,
> 
> I believe we can use Google Meet and plan it for later because of my heavy workload.

@LeaveNhA

Understood. Also just letting you know in advance that as an introvert, my communication skills can be lacking at times, and I’ve never been in a proper meeting before, so I’m asking for some patience in advance. I am willing to go out of my comfort zone for the sake of project progression, so do let me know when the time comes.

**Comment by Chamotrans** [comment]
> 

Sorry but could you please fix URL again as it is not inaccessible.

**Comment by bslatyer** [comment]
Hi,

Here's the new link to download the IPA

https://www.dropbox.com/scl/fi/8p1e9940gmfmj56n9mhvo/OpenParsec.ipa?rlkey=ui119yrx7lzfc38yuxd5qtx3j&dl=0

Thanks,

Braeden

**Comment by MaximeCrp** [comment]
@bslatyer Thank you for the build. Is it supposed to contain the keyboard commits? Could not make it work. Tested on a M1 iPad with a BT MX Keys and a logitech cover keyboard. Parsec does not receive keyboard inputs

**Comment by LeaveNhA** [comment]
> @bslatyer Thank you for the build. Is it supposed to contain the keyboard commits? Could not make it work. Tested on a M1 iPad with a BT MX Keys and a logitech cover keyboard. Parsec does not receive keyboard inputs

Yes, aware of that. I'm working on a fix. Thank you for your patience.

**Comment by MaximeCrp** [comment]
Thank you! I was just wondering if it was just this build but building the app myself worked flawlessly and same result indeed.
Good luck to you and Angel building this wonder

**Comment by maxdluffy** [comment]
Hi, I really appreciate the open parsec project and have enjoyed using the app this past month. 
But now suddenly there are issues that I don't know how to fix. 
1. As soon as the app connects to the internet, it crashes.
It doesn't crash with wifi off. And it doesn't crash when signing in. Only in the main screen. 
2. Scarlet is unable to install a fresh copy anymore. It just says failed to install when I tap on the icon of this app. 
I REAALY would appreciate any help for this please.

**Comment by AngelDTF** [comment]
It appears Parsec changed something with their API. I've got other things running on my primary machine right now, so if I'm not busy later today, I'll clone the project to my laptop and see if I can diagnose the issue. Thanks for the heads up.

**Comment by LeaveNhA** [comment]
Hey @AngelDTF, did you debug it? I have a product to deliver, I can't catch-up with you right now, sorry.

**Comment by AngelDTF** [comment]
No worries, I was able to figure out the problem. The API was no longer sending some information the app was expecting, but the info wasn’t necessary for making the app functional, so I just commented out the lines requiring it, and it appears to be working perfectly fine now.

**Comment by maxdluffy** [comment]
So, how will I be able to use it again?
Will you upload a new .ipa?

**Comment by LeaveNhA** [comment]
> No worries, I was able to figure out the problem. The API was no longer
 sending some information the app was expecting, but the info wasn’t
 necessary for making the app functional, so I just commented out the lines
 requiring it, and it appears to be working perfectly fine now.

You are the best @AngelDTF!

**Comment by AngelDTF** [comment]
> So, how will I be able to use it again?
> Will you upload a new .ipa?
@maxdluffy 
I'm not in the Apple Developer Program, so I myself cannot distribute IPAs, sorry. You'll have to ask someone who's in the program to upload a new build.

**Comment by bslatyer** [comment]
I will build it later today

**Comment by maxdluffy** [comment]
Thank you very much!

**Comment by bslatyer** [comment]
It's now been complied and uploaded using the same Dropbox link

**Comment by Androlefas** [comment]
The link doesnt work for me, it say i cant access to the link, could you fix it? Big thanks !

**Comment by bslatyer** [comment]
> Hi,
> 
> Here's the new link to download the IPA
> 
> https://www.dropbox.com/scl/fi/8p1e9940gmfmj56n9mhvo/OpenParsec.ipa?rlkey=zutdvgzmqkeif1zcr7cgo5x2f&dl=0
> 
> Thanks,
> 
> Braeden

@Androlefas it's fixed.

**Comment by Androlefas** [comment]
Hello,

Thank you ! The link is working now.

I have a another issue, when i try to install it, i have the error that say the integrity cant be verify. I have a developer account.

Tank you again for your help

**Comment by bslatyer** [comment]
> Hello,
> 
> Thank you ! The link is working now.
> 
> I have a another issue, when i try to install it, i have the error that say the integrity cant be verify. I have a developer account.
> 
> Tank you again for your help

You need to sign this IPA with your own Apple Distribution (iOS) and associated provisioning profile.

As you and your devices are not on my Development Team - this will not work with your devices unless you resign it yourself.

The builds are "as is" and I won't give support unless the build is completely broken.

Thanks,

Braeden

**Comment by Androlefas** [comment]
Hello,

i understand now. Thank you again for these explanations.

**Comment by bslatyer** [comment]
Hey y'all,

I've added a backup link for the latest IPA build in case Dropbox decides to go AWOL.

[OpenParsec](https://assets.slats.au/ios/OpenParsec.ipa)

Thanks,

Braeden

**Comment by maxdluffy** [comment]
Hi, Would it be possible to somehow use my secondary screen on Open parsec without making it the main screen?
There is a feature for this on the Parsec android app. I would really appreciate it if it can be implemented.  Or if there's an alternate way of doing it that you can let me know of.

**Comment by maxdluffy** [comment]
Never mind me, the host settings allow you to do that anyway, It just didn't work for me before, because there's a bug where you have to choose some other screen and then select your favored screen again every other time. Really appreciate this project though. God bless.

**Comment by AngelDTF** [comment]
I want to see if maybe I can reimplement the switch screens button via some reverse engineering, but no guarantees as it doesn't seem to be a part of the old SDK.

Regardless though, this issue has gone pretty off-topic, so I'll be closing it here. Thanks everyone for the support on the project. I really want to contribute more towards this project once I have some more free time on my hands, and hopefully get it into a more functional state.

**Comment by dmt-hub** [comment]
The dropbox and other link are not working anymore. Where can i find a new link? Thank you.

**Comment by bslatyer** [comment]
@dmt-hub this has been fixed now. Please use this [link](https://assets.slats.au/ios/OpenParsec.ipa) to download it.

**Comment by dmt-hub** [comment]
Thank you!

**Comment by carvacs** [comment]
Can i have access to the ipa file? I enter the link you provide but i not authorised @dmt-hub.

**Comment by niandralades** [comment]
@bslatyer please give access for ipa.

jordanrobertwebb@gmail.com

**Comment by bslatyer** [comment]
Hey @niandralades & @carvacs 

You can the latest IPA here at https://github.com/bslatyer/OpenParsec/releases/tag/1.0

Thanks,

Braeden

**Comment by Kodikuu** [comment]
@bslatyer Per Parsec's ToS, you may not provide IPAs to users. Users must compile their own IPA

**Comment by Kodikuu** [comment]
More specifically; You may not provide a compiled binary to anyone that uses email+password or session auth. The only allowed form is now defunct, so you just _can't_ provide them
